host.go

```diff
@@ -0,0 +1,372 @@
+package httpwasm
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/textproto"
+	"net/url"
+	"sort"
+	"strings"
+
+	"github.com/http-wasm/http-wasm-host-go/api/handler"
+	handlerapi "github.com/http-wasm/http-wasm-host-go/api/handler"
+	"github.com/labstack/echo/v4"
+)
+
+type requestStateKey struct{}
+
+type requestState struct {
+	c        echo.Context
+	next     echo.HandlerFunc
+	features handlerapi.Features
+}
+
+func newRequestState(c echo.Context, next echo.HandlerFunc, features handlerapi.Features) *requestState {
+	s := &requestState{c, next, features}
+	s.enableFeatures(features)
+	return s
+}
+
+func (s *requestState) enableFeatures(features handlerapi.Features) {
+	// TODO(jcchavezs): enable features
+}
+
+func (s *requestState) handleNext() (err error) {
+	defer func() {
+		if recovered := recover(); recovered != nil {
```

Comment:
in nethttp we recover because ServeHTTP is documented to panic. 

// If ServeHTTP panics, the server (the caller of ServeHTTP) assumes
// that the effect of the panic was isolated to the active request.

We are missing a comment on nethttp impl about that being why we rescue. Can you comment here if indeed it is documented that recovering is safe, or don't recover if it isn't?